infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -63,33 +92,22 @@
     private final ConcurrentHashMap<UUID, Long> streamTailToGlobalTailMap = new ConcurrentHashMap<>();
 
     /**
-     * A cache of conflict parameter modification timestamps.
+     * A cache that records recent (stream, conflict parameter) modification
+     * timestamps (global addresses).
      *
-     * A "wildcard" {@link SequencerServer::maxConflictWildcard} representing all the entries which were evicted from the cache holds
-     * theier maximal modification time.
+     * And a "wildcard" representing all the entries which were evicted from the cache
+     * holds their maximal modification time.
      */
     private final long maxConflictCacheSize = 10_000;
     private long maxConflictWildcard = -1L;
-    private final Cache<Integer, Long> conflictToGlobalTailCache = Caffeine.newBuilder()
+    private final Cache<AbstractMap.SimpleEntry<Integer,Integer>, Long>
```

Comment:
It's too hard to understand. I'm going to have to object. It's hard to figure out why its a simpleEntry, and the fact that it's integer,integer, doesn't help.

You need to create a new class to hold this data and override hashCode to provide a hashCode which generates a hash that combines UUID and conflictParam. The @Data lombok annotation is handy for this.